net-libs/mbedtls/mbedtls-2.6.0.recipe

```diff
@@ -0,0 +1,118 @@
+LICENSE="Apache v2"
+REVISION="1"
+SOURCE_URI="https://tls.mbed.org/download/mbedtls-$portVersion-apache.tgz"
+CHECKSUM_SHA256="99bc9d4212d3d885eeb96273bcde8ecc649a481404b8d7ea7bb26397c9909687"
+
+ARCHITECTURES="?x86_gcc2 ?x86 x86_64"
+
+PROVIDES="
+	mbedtls = $portVersion
+	cmd:mbedtls_aescrypt2 = $portVersion
+	cmd:mbedtls_benchmark = $portVersion
+	cmd:mbedtls_cert_app = $portVersion
+	cmd:mbedtls_cert_req = $portVersion
+	cmd:mbedtls_cert_write = $portVersion
+	cmd:mbedtls_crl_app = $portVersion
+	cmd:mbedtls_crypt_and_hash = $portVersion
+	cmd:mbedtls_dh_client = $portVersion
+	cmd:mbedtls_dh_genprime = $portVersion
+	cmd:mbedtls_dh_server = $portVersion
+	cmd:mbedtls_dtls_client = $portVersion
+	cmd:mbedtls_dtls_server = $portVersion
+	cmd:mbedtls_ecdh_curve25519 = $portVersion
+	cmd:mbedtls_ecdsa = $portVersion
+	cmd:mbedtls_gen_entropy = $portVersion
+	cmd:mbedtls_gen_key = $portVersion
+	cmd:mbedtls_gen_random_ctr_drbg = $portVersion
+	cmd:mbedtls_gen_random_havege = $portVersion
+	cmd:mbedtls_generic_sum = $portVersion
+	cmd:mbedtls_hello = $portVersion
+	cmd:mbedtls_key_app = $portVersion
+	cmd:mbedtls_key_app_writer = $portVersion
+	cmd:mbedtls_mini_client = $portVersion
+	cmd:mbedtls_mpi_demo = $portVersion
+	cmd:mbedtls_pem2der = $portVersion
+	cmd:mbedtls_pk_decrypt = $portVersion
+	cmd:mbedtls_pk_encrypt = $portVersion
+	cmd:mbedtls_pk_sign = $portVersion
+	cmd:mbedtls_pk_verify = $portVersion
+	cmd:mbedtls_req_app = $portVersion
+	cmd:mbedtls_rsa_decrypt = $portVersion
+	cmd:mbedtls_rsa_encrypt = $portVersion
+	cmd:mbedtls_rsa_genkey = $portVersion
+	cmd:mbedtls_rsa_sign = $portVersion
+	cmd:mbedtls_rsa_sign_pss = $portVersion
+	cmd:mbedtls_rsa_verify = $portVersion
+	cmd:mbedtls_rsa_verify_pss = $portVersion
+	cmd:mbedtls_selftest = $portVersion
+	cmd:mbedtls_ssl_cert_test = $portVersion
+	cmd:mbedtls_ssl_client1 = $portVersion
+	cmd:mbedtls_ssl_client2 = $portVersion
+	cmd:mbedtls_ssl_fork_server = $portVersion
+	cmd:mbedtls_ssl_mail_client = $portVersion
+	cmd:mbedtls_ssl_server = $portVersion
+	cmd:mbedtls_ssl_server2 = $portVersion
+	cmd:mbedtls_strerror = $portVersion
+	cmd:mbedtls_udp_proxy = $portVersion
+	lib:libmbedcrypto
+	lib:libmbedtls
+	lib:libmbedx509
+	"
+REQUIRES="
+	haiku
+	"
+
+PROVIDES_devel="
+	mbedtls_devel = $portVersion
+	devel:libmbedcrypto
+	devel:libmbedtls
+	devel:libmbedx509
+	"
+REQUIRES_devel="
+	mbedtls == $portVersion base
+	"
+
+BUILD_PREREQUIRES="
+	cmd:gcc
+	cmd:make
+	cmd:perl
+	cmd:sed
+	"
+BUILD_REQUIRES="
+	haiku_devel
+	"
+
+PATCH()
+{
+	sed -i \
+	-e "s|/include/mbedtls|/$relativeOldIncludeDir|;"\
+	-e "s|/include|/$relativeOldIncludeDir|;"\
+	-e "s|/bin/|/$relativeBinDir/|;"\
+	-e "s|/bin\$|/$relativeBinDir/|;"\
+	-e "s|/lib\$|$relativeLibDir/|;"\
+	Makefile
+}
+
+BUILD()
+{
+	# Haiku is unix-like and network functions are in libnetwork
+	make CFLAGS="-Dunix" LDFLAGS="-lnetwork" SHARED=yes WARNING_CFLAGS=
+}
+
+INSTALL()
+{
+	make install SHARED=yes DESTDIR=$prefix/
```

Comment:
It worked for my without the / after $prefix.
Could you try removing the / ?

Reply:
I tried, it tried writing to `/packages/mbedtls-2.6.0-1/.selflib` instead of `/packages/mbedtls-2.6.0-1/.self/lib`

Reply:
Perhaps we are running different haikuporter versions? I am on `969cac0bb71b...`

Reply:
I'm running a quite recent hrev on both x86_64 and x86_gcc2, but your issue about the $prefix/ does not comme from the version of Haiku.
The default value does not have a trailing slash:
```
grep ^DESTDIR Makefile 
DESTDIR=/usr/local
```
If you trie to clean the work dir with
```
haikuporter -c mbedtls
```
and then try to build it again, do you still get the error?
I have an "hp" alias for haikuporter which is set to:
```
haikuporter -G --get-dependencies --no-source-packages
```
And when I call `hp -S mbedtls` I don't have any ossues.

Reply:
I have tried deleting my work directory, still nothing... I will try to delete everything and run `haikuporter -c mbedtls` and try again.